.github/workflows/doc-tests.yml

```
@@ -0,0 +1,33 @@
+name: test-docs
+run-name: Documentation tests
+
+on:
+  pull_request:
+  push:
+    branches: [master]
+
+permissions:
+  contents: read
+
+jobs:
+  doc-test:
+    name: Sphinx-${{ matrix.builder }}
+    runs-on: ubuntu-latest
+    continue-on-error: ${{ matrix.experimental || false }}
+    strategy:
+      fail-fast: false
+      matrix:
+        builder: [ linkcheck, doctest, html ]
+        include:
+          # Run default html builder with warnings as error
+          - builder: html
+            args: -W
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.x
+      - name: Install the project and docs dependencies
+        run: pip install -e .[docs]
+      - name: Run sphinx builder ${{ matrix.builder }}
+        run: sphinx-build -b ${{ matrix.builder }} ${{ matrix.args }} ./docs ./docs/_build
```

@martinhoyer I was considering this 
```suggestion
      - name: Run sphinx builder ${{ matrix.builder }}
        run: |
          hatch run docs:${{ matrix.builder }} ${{ matrix.args }}
```
But I can't seem to pass arguments like `-W`. Any advice on this?

---

I think the confusion arises from mixing "docs" optional extras deps and "docs" hatch env deps.  

I believe @happz is the right person to do a review here.

---

For this I was trying to unify the script environments and since `hatch-env` is already adapted, it should be using that. But I still don't understand the hatch scripts. Did you see any implementation where they passed more arguments to `hatch run`?

---

Oh found it. I had to add `{args}`